addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java

```diff
@@ -521,6 +522,76 @@ public void loadRssiValues() throws IOException {
         }
     }
 
+    @Override
+    public void setInstallMode(boolean enable, int seconds) throws IOException {
+        HmDevice gwExtrasHm = devices.get(HmDevice.ADDRESS_GATEWAY_EXTRAS);
+        
+        if (gwExtrasHm != null) {
+            // since the homematic virtual device exist: try setting install mode via its dataPoints
+            HmDatapoint installModeDataPoint = null;
+            HmDatapoint installModeDurationDataPoint = null;
+
+            // collect virtual datapoints to be accessed
+            HmChannel hmChannel = gwExtrasHm.getChannel(HmChannel.CHANNEL_NUMBER_EXTRAS);            
+            HmDatapointInfo installModeDurationDataPointInfo = new HmDatapointInfo(HmParamsetType.VALUES, hmChannel,
+                    HomematicConstants.VIRTUAL_DATAPOINT_NAME_INSTALL_MODE_DURATION);
+            if (enable) {
+                installModeDurationDataPoint = hmChannel.getDatapoint(installModeDurationDataPointInfo);
+            }
+
+            HmDatapointInfo installModeDataPointInfo = new HmDatapointInfo(HmParamsetType.VALUES, hmChannel,
+                    HomematicConstants.VIRTUAL_DATAPOINT_NAME_INSTALL_MODE);
+            
+            installModeDataPoint = hmChannel.getDatapoint(installModeDataPointInfo);
+                        
+            // first set duration on the datapoint
+            if (installModeDurationDataPoint != null) {
+                try {
+                    VirtualDatapointHandler handler = getVirtualDatapointHandler(installModeDurationDataPoint, null);
+                    handler.handleCommand(this, installModeDurationDataPoint, new HmDatapointConfig(), seconds);
+                    
+                    // notify thing if exists 
+                    gatewayAdapter.onStateUpdated(installModeDurationDataPoint);
+                } catch (HomematicClientException ex) {
+                    logger.error("Failed to send datapoint {}", installModeDurationDataPoint, ex);
```

Comment:
Just to be sure: We can only end up here if there is something seriously wrong (i.e. it can be considered a software bug) and it is a non-recoverable situation, correct?

Reply:
It is very unlikely to happen (if the datapoint is available, the command should be handled without problems), but it is not a non-recoverable situation. Therefore I will change it to warning here.